checker.cpp

```diff
@@ -1,17 +1,126 @@
 #include <assert.h>
+#include<iostream>
+#include<vector>
+using namespace std;
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
+class Alert
+{
+  public:
+  virtual void sendAlert(const string vitalName,const char* message) = 0;
+  
+};
+
+class AlertInSms: public Alert
+{
+  public:
+  void sendAlert(const string vitalName,const char* message)
+  {
+    std::cout<<"SMS sent - "<<vitalName<<" "<<message<<std::endl;
+  }
+};
+
+class AlertinSound: public Alert
+{
+  public:
+  void sendAlert(const string vitalName,const char* message)
+  {
+    std::cout<<"Alarm Sound - "<<vitalName<<" "<<message<<std::endl;
+  }
+};
+
+struct vital
+{
+  string vitalName;
+  int lowerLimit,upperLimit,vitalValue;
+  vital(string name,int lower,int upper)
+  {
+    vitalName = name;
+    lowerLimit = lower;
+    upperLimit = upper;
+  }
+};
+
+bool vitalIsOk(Alert *alert, const string vitalName,float value,int lower,int upper)
+{
+  if(value < lower)
+  {
+    alert->sendAlert(vitalName,"is low");
     return false;
-  } else if(respRate < 30 || respRate > 95) {
+  }
+  else if(value > upper)
+  {
+    alert->sendAlert(vitalName,"is high");
     return false;
   }
   return true;
+  
+}
+
+bool vitalsAreOk(Alert* alert,vector<vital>vitalsList) {
+  bool check = true;
+  
+  for(int i=0;i<vitalsList.size();i++)
+  {
+    if(!vitalIsOk(alert,vitalsList[i].vitalName,vitalsList[i].vitalValue,vitalsList[i].lowerLimit,vitalsList[i].upperLimit))
+      check=false;
+  }
+  
+  return check;
+}
+
+void setVitalValue(vector<vital>&vitalsList,string name,int value)
+{
+  for(int i=0;i<vitalsList.size();i++)
+  {
+    if(vitalsList[i].vitalName.compare(name) == 0)
+      vitalsList[i].vitalValue = value;
```

Comment:
Nice use of containers 👍 
When you need to search through the vector, consider using a map. In this case, it could be a map of vital name to its value etc.

Reply:
If we separate the vital value from the structure of vital and create the map of vital name to its value then we need to search through the vector to find the vital's corresponding lower and upper limits. 

In case if we want to eliminate the searching process, the approach we can use is create separate maps as follows:
Vital : Lower limit
Vital : Upper limit
Vital : Value
We can directly access vital corresponding data directly.